src/librustdoc/lib.rs

```diff
@@ -252,8 +276,33 @@ pub fn main_args(args: &[String]) -> isize {
     let markdown_input = input.ends_with(".md") || input.ends_with(".markdown");
 
     let output = matches.opt_str("o").map(|s| PathBuf::from(&s));
+    let css_file_extension = matches.opt_str("e").map(|s| PathBuf::from(&s));
     let cfgs = matches.opt_strs("cfg");
 
+    // we now check if unstable options are allowed and if we're in a nightly build
+    let mut has_z_unstable_options = false;
+    for flag in matches.opt_strs("Z").iter() {
+        if *flag != "unstable-options" {
```

Comment:
I'm not sure about what @brson wants in here. He said some other options than "unstable-options" could be added so I wrote my code in this direction. But for now I'll just do it the simplest way.